security/src/main/java/io/micronaut/security/expressions/SecuredEvaluationContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.expressions;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.security.authentication.Authentication;
+import io.micronaut.security.utils.SecurityService;
+import jakarta.inject.Singleton;
+
+/**
+ * Context for supporting {@link io.micronaut.security.annotation.Secured} annotation expressions with object references.
+ */
+@Singleton
+public class SecuredEvaluationContext {
```

Comment:
annotate with `@Internal` and make `final`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-security/pull/1309/commits/6935e79452ba1aa1f8f6dc3aad517781f0e1c231